src/Integration.UnitTests/Binding/CSharpVBBindingOperationTests.cs

```diff
@@ -396,13 +396,74 @@ public void ProjectBindingOperation_Prepare_NoAdditionalFileConflict_NoException
             additionalFileConflictChecker.VerifyAll();
         }
 
+        [TestMethod]
```

Comment:
Because we use mocks of VS project, I'm limited with the cases I can test.
For example, I couldn't test "when project has no property specified" -- since VS will return a default value anyway, and our mocks will not.
I also couldn't test that if the user didn't specify a property, then VS will unite Debug/Release into a single non-conditional property.

Reply:
I agree we don't need to test case where VS unites the Debug/Release properties - we're only interested in testing that our code correctly updates properties for all of the current configurations.

I'm not sure about your first point, though - it's our mock, so we can change its behaviour as necessary. Having said that, I'm not clear which cases the mock won't handle. What can't be tested - the case when the property information map is empty?

Reply:
The case where we set no project properties, yet the property information map is not empty. But it's a minor case.

Reply:
But the property information map will only contain a value because VS has supplied a default property, no? Which we can simulate by having the test create a property that returns the default value.

From a test point of view, the cases we might want to cover are:
1) property information map is empty
2) property information map has default value(s)
3) property information map ha non-default value(s)
Can we test all of those?

Reply:
I tested 2+3, the first one is not realistic, as it will never be empty. So I think it's ok.